### ISSUE TITLE
Installing plugins from github branch stopped working on 1.55.0

### ISSUE THREAD

**Issue by Katafalkas** [serverless/serverless#6884]
# Bug Report

## Description

1. What did you do?
Trying to install plugin from github branch in `package.json`:
works fine:
```
devDependencies: {
  "serverless-domain-manager": "^3.0.0"
}
```
stopped working in `1.55.0`:
```
devDependencies: {
  "serverless-domain-manager": "Katafalkas/serverless-domain-manager#js-artifacts",
}
```
does not work without js artifacts either:
```
devDependencies: {
  "serverless-domain-manager": "Katafalkas/serverless-domain-manager#feature/support-nested-stacks",
}
```
1. What happened?
I get an error:
```
ServerlessError: Serverless plugin "serverless-domain-manager" not found. Make sure it's installed and listed in the "plugins" section of your serverless config file.
```
1. What should've happened?
Should not have gotten an error. Did not get this error before `1.55.0`
1. What's the content of your `serverless.yml` file?
```
service: support-api
frameworkVersion: ">=v1.28"
custom:
  ...
provider:
  name: aws
  runtime: nodejs10.x
  region: ${self:custom.cfg.region}
  stage: ${self:custom.cfg.stage}
  iamRoleStatements: ${file(environments/iamRoleStatements.yml):iamRoleStatements}
  logRetentionInDays: 14
  memorySize: 256
  timeout: 120
  environment:
    ...
package:
  ...
plugins:
  - serverless-sentry
  - serverless-plugin-include-dependencies
  - serverless-domain-manager
  - serverless-plugin-split-stacks

resources:
  ...
functions:
  ...
```
1. What's the output you get when you use the `SLS_DEBUG=*` environment variable (e.g. `SLS_DEBUG=* serverless deploy`)
```
$ SLS_DEBUG=* sls deploy
  Serverless Error ---------------------------------------
 
  ServerlessError: Serverless plugin "serverless-domain-manager" not found. Make sure it's installed and listed in the "plugins" section of your serverless config file.
      at pluginsObject.modules.filter.map.name (/Users/audriuskucinskas/.nvm/versions/node/v10.16.3/lib/node_modules/serverless/lib/classes/PluginManager.js:138:17)
      at Array.map (<anonymous>)
      at PluginManager.resolveServicePlugins (/Users/audriuskucinskas/.nvm/versions/node/v10.16.3/lib/node_modules/serverless/lib/classes/PluginManager.js:123:8)
      at PluginManager.loadAllPlugins (/Users/audriuskucinskas/.nvm/versions/node/v10.16.3/lib/node_modules/serverless/lib/classes/PluginManager.js:110:15)
      at pluginManager.loadConfigFile.then.then (/Users/audriuskucinskas/.nvm/versions/node/v10.16.3/lib/node_modules/serverless/lib/Serverless.js:78:35)
  From previous event:
      at Serverless.init (/Users/audriuskucinskas/.nvm/versions/node/v10.16.3/lib/node_modules/serverless/lib/Serverless.js:76:8)
      at initializeErrorReporter.then (/Users/audriuskucinskas/.nvm/versions/node/v10.16.3/lib/node_modules/serverless/bin/serverless.js:71:8)
      at runCallback (timers.js:705:18)
      at tryOnImmediate (timers.js:676:5)
      at processImmediate (timers.js:658:5)
      at process.topLevelDomainCallback (domain.js:126:23)
  From previous event:
      at Object.<anonymous> (/Users/audriuskucinskas/.nvm/versions/node/v10.16.3/lib/node_modules/serverless/bin/serverless.js:61:4)
      at Module._compile (internal/modules/cjs/loader.js:778:30)
      at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
      at Module.load (internal/modules/cjs/loader.js:653:32)
      at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
      at Function.Module._load (internal/modules/cjs/loader.js:585:3)
      at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
      at startup (internal/bootstrap/node.js:283:19)
      at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              10.16.3
     Framework Version:         1.55.1
     Plugin Version:            3.2.0
     SDK Version:               2.1.2
     Components Core Version:   1.1.2
     Components CLI Version:    1.4.0
```

Similar or dependent issues:

- Not found

**Comment by Katafalkas** [comment]
After digging around, still did not resolve the issue, but found a "hacky" solution.

If I manually place the plugin into projects `.serverless_plugins` with js artifacts in it - plugin works.

So currently the issue is that after `1.55.0` my plugin can not be imported if it is only in `./node_modules`.

**Comment by medikoo** [comment]
I've looked into this issue, and it's indirectly caused by malformed configuration of _package.json_ `main` in `Katafalkas/serverless-domain-manager#js-artifacts`, as it points to [non existing `dist/index.js` file](https://github.com/Katafalkas/serverless-domain-manager/blob/7469d18376adeed9934462fdf790d4e8fe20468b/package.json#L27)

Still, direct cause is that resolver ([`ncjsm/resolve/sync`](https://github.com/medikoo/ncjsm#resolvesyncdir-path)) we use to resolve plugin paths against service path, just stops on invalid `main`, while module resolution as it happens in Node.js  natively, fallbacks in such case to `index.js` in package directory (and as `index.js` exists in `Katafalkas/serverless-domain-manager#js-artifacts` issue is not exposed).

I've just fixed this issue in `ncjsm`, and published with [v4.0.1](https://github.com/medikoo/ncjsm/releases/tag/v4.0.1), so after running `npm update -g serverless` issue should be fixed on your side.

**Comment by medikoo** [comment]
Closing, as I believe it was addressed
